client/client-lib/src/mint/backup.rs

```diff
@@ -216,5 +330,393 @@ impl EcashBackup {
     }
 }
 
+/// The state machine used for fast-fowarding backup from point when it was taken to the present time.
+struct EcashRecoveryNonceTracker {
+    /// Nonces that we track that are currently spendable.
+    spendable_note_by_nonce: HashMap<Nonce, (Amount, SpendableNote)>,
+
+    /// Outputs (by `OutPoint`) we track federation member confirmations for blind nonces.
+    ///
+    /// Once we get enough confirmation (valid shares), these become new spendable notes.
+    ///
+    /// Note that `NoteIssuanceRequest` is optional, as sometimes we might need
+    /// to handle a tx where only some of the blind nonces were in the pool.
+    /// A `None` means tha this blind nonce/message is there only for validation
+    /// purposes, and will actually not create a `spendable_note_by_nonce`
+    #[allow(clippy::type_complexity)]
+    pending_outputs: HashMap<
+        OutPoint,
+        (
+            TieredMulti<(BlindedMessage, Option<NoteIssuanceRequest>)>,
+            HashMap<PeerId, Vec<tbs::BlindedSignatureShare>>,
+        ),
+    >,
+
+    /// Next nonces that we expect might soon get used.
+    /// Once we see them, we move the tracking to `pending_outputs`
+    ///
+    /// Note: since looking up nonces is going to be the most common operation
+    /// the pool is kept shared (so only one lookup is enough), and replenishment
+    /// is done each time a note is consumed.
+    pending_nonces: HashMap<BlindedMessage, (NoteIssuanceRequest, NoteIndex, Amount)>,
```

Comment:
I wonder if this should be a `HashMap<(BlindedMessage, Amount), (NoteIssuanceRequest, NoteIndex)>` instead? That would remove the failure case of "wrong amount" because that would just become "nonce isn't ours, ignore".

Reply:
That would work, though I wanted to handle this failure case, just so we can log it. Now sure how useful it is. It seems important, but then user's can't really do anything about it, and they will rarely see it anyway.